src/ens.ts

```diff
@@ -186,6 +185,49 @@ export default class Ens extends EthereumNamingService {
     };
   }
 
+  /**
+   * resolves an ipfsHash stored on domain
+   * @param domain - domain name
+   */
+  async ipfsHash(domain: string): Promise<string> {
+    const hash = await this.getContentHash(domain);
+    if (!hash)
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        recordName: 'ipfs hash',
+        domain: domain,
+      });
+    return hash;
+  }
+
+  /**
+   * resolves a httpUrl stored on domain
+   * @param domain - domain name
+   */
+  async httpUrl(domain: string): Promise<string> {
+    const resolver = await this.getResolverContract(domain);
+    const httpUrl = await this.getTextRecord(resolver, domain, 'url');
+    if (!httpUrl)
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        recordName: 'redirect url',
+        domain: domain,
+      });
+    return httpUrl;
+  }
+  /**
+   * resolves an email stored on domain
+   * @param domain - domain name
+   */
+  async email(domain: string): Promise<string> {
+    const resolver = await this.getResolverContract(domain);
+    const email = await this.getTextRecord(resolver, domain, 'email');
+    if (!email)
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        recordName: 'email',
+        domain: domain,
+      });
```

Comment:
The code is copypasted between methods. you clean it up by moving the `throw` part into `getTextRecord`.